fsharp-backend/src/HttpMiddleware/HttpMiddlewareV1.fs

```diff
@@ -0,0 +1,134 @@
+/// Thin, bytes-friendly middleware
+///
+/// This Middleware corresponds with `HTTPBytes` handlers
+[<RequireQualifiedAccess>]
+module HttpMiddleware.HttpMiddlewareV1
+
+module Request =
+  open Prelude
+  open LibExecution.VendoredTablecloth
+
+  module RT = LibExecution.RuntimeTypes
+  module HttpQueryEncoding = BackendOnlyStdLib.HttpQueryEncoding
+
+  let private parseQueryString (query : string) : RT.Dval =
+    // Drop leading "?"
+    let query = if query.Length > 0 then String.dropLeft 1 query else query
+    HttpQueryEncoding.ofFormEncoding query
+
+  let private parseHeaders (headers : (string * string) list) =
+    headers
+    |> List.map (fun (k, v) -> (String.toLowercase k, RT.DStr v))
```

Comment:
Should headers be a string here (or bytes)? Should we lowercase the value too as well as the key?

Reply:
I'm not sure why they would be bytes. As far as I know, and have just googled, headers (both keys and values) may only be strings.

I'm not sure why we'd lowercase either key _or_ value - I'd lean towards removing that existing `toLowercase`. Why should we do such for either?

Reply:
The reason i was asking are the strings and Dark are Unicode, but I didn't expect that Unicode would be supported and HTTP headers. It looks like only printable ASCII characters are allowed, so I guess using strings is the best way to support this.

The reason to use lowercase letters is because http2 and to http3 both use lowercase headers. Since the user is likely to do a string comparison, or possibly even a map look up, they can get different answers depending on whether the request was made with http1 vs http2/3. I observed this with the old http framework, and I'd like to avoid it with the new one.

Reply:
> The reason to use lowercase letters

Ah, that's great context, thanks!

Reply:
Should the _response_ headers also be to-lowercase'd, then?

Reply:
While the keys are case-insensitive (per the spec), the values are not - so let's lowercase both request header keys on the way in, and response header keys on the way out.